src/util/invariant.h

```diff
@@ -196,6 +199,9 @@ std::string get_backtrace();
 
 void report_exception_to_stderr(const invariant_failedt &);
 
+/// This class is the backbone of all the invariant types. Every instance of an
+/// invariant is ultimately generated by this class, which is at times called
+/// via a wrapper function.
```

Comment:
I am not sure this comment is in the right place? It's now part of the documentation of a method, not a class?

Reply:
Yes - whoops! Function.